common/archiver/gcloud/visibilityArchiver.go

```diff
@@ -162,8 +163,8 @@ func (v *visibilityArchiver) Query(
 	saTypeMap searchattribute.NameTypeMap,
 ) (*archiver.QueryVisibilityResponse, error) {
 
-	if err := v.ValidateURI(URI); err != nil {
-		return nil, &serviceerror.InvalidArgument{Message: archiver.ErrInvalidURI.Error()}
+	if strings.TrimSpace(request.Query) == "" {
+		return v.queryAll(ctx, URI, request, saTypeMap)
 	}
```

Comment:
It's not just that--the query itself needs to be different

Reply:
I don't follow. I original thought you want to validate query first, so we don't have to make a separate call for validating URI if the query is invalid. Is my understanding correct, or you're trying to achieve something else?

Also, I think other impl (filestore & s3) should get the same change ^?

Reply:
The gcloud and s3 backends must branch their logic because they use a different SQL query which no longer contains a where clause. All three backends are okay with empty queries and use the same validation.

Reply:
I think I understand ^. I am just confused about how it relates to the process of validating URI. Maybe we can sync offline.

Reply:
^ was a mistake. The URI validation should not be changed. Reverted that aspect.